test/logflare/backends/adaptor/clickhouse_adaptor/pipeline_test.exs

```diff
@@ -0,0 +1,196 @@
+defmodule Logflare.Backends.Adaptor.ClickhouseAdaptor.PipelineTest do
+  use Logflare.DataCase
+  import Mimic
+
+  alias Broadway.Message
+  alias Logflare.Backends.Adaptor.ClickhouseAdaptor
+  alias Logflare.Backends.Adaptor.ClickhouseAdaptor.Pipeline
+  alias Logflare.Backends.Adaptor.ClickhouseAdaptor.ConnectionManager
+  alias Logflare.Backends.BufferProducer
+  alias Logflare.Backends.SourceRegistry
+  alias Logflare.LogEvent
+
+  setup :verify_on_exit!
+
+  describe "child_spec/1" do
+    test "returns proper child specification" do
+      spec = Pipeline.child_spec(:some_arg)
+
+      assert spec.id == Pipeline
+      assert spec.start == {Pipeline, :start_link, [:some_arg]}
+    end
+  end
+
+  describe "start_link/1" do
+    setup do
+      user = insert(:user)
+      source = insert(:source, user: user)
+      backend = insert(:backend, type: :clickhouse)
+
+      adaptor_state = %ClickhouseAdaptor{
+        source: source,
+        backend: backend,
+        pipeline_name: {:via, Registry, {SourceRegistry, {source.id, backend.id, Pipeline}}},
+        connection_name: {:via, Registry, {SourceRegistry, {source.id, backend.id, :connection}}}
+      }
+
+      %{adaptor_state: adaptor_state}
+    end
+
+    test "calls Broadway.start_link with correct configuration", %{adaptor_state: adaptor_state} do
+      Broadway
+      |> expect(:start_link, fn module, opts ->
+        assert module == Pipeline
+        assert opts[:name] == adaptor_state.pipeline_name
+        assert opts[:context] == adaptor_state
+
+        # Verify producer configuration
+        producer_config = opts[:producer]
```

Comment:
hmm do we need to assert configuration though? i think this seems somewhat redundant and verges on testing the library

Reply:
would lean towards conciseness even if this is for coverage.